### PR TITLE
Fix out-of-bounds heap access during metadata encoding from PNG

### PR DESCRIPTION
**Pull request by fherzog2**
Scenario:
- convert JPG to PNG with image magick (to preserve any metadata in the JPG)
- convert PNG to FLIF
- extract metadata from FLIF using the API
- **metadata size does not match size in JPG**

This happens because flif is storing the wrong length for the metadata chunk, namely the length it has in PNG, where the chunk is hex-encoded as text. So flif metadata gets encoded with kbytes of garbage at the end of the chunk.

### CODE REVIEW

File: src/image/image-png-metadata.hpp [hunk 1]
```diff
@@ -39,8 +39,9 @@ static unsigned char* HexStringToBytes(const char* hexstring,
   return raw_data;
 }
 
+// rewrite
```
Comment (bjorn3) [inline src/image/image-png-metadata.hpp]: Why did you add this comment?

Comment (fherzog2) [inline src/image/image-png-metadata.hpp]: Good catch, don't know why I overlooked that.

**Comment by jonsneyers** [comment]
Thanks for the fix!
